### PR TITLE
Replace deprecated APIs

### PR DESCRIPTION
**Pull request by hugueskamba**
Reviewers
@paul-szczepanek-arm @pan- 

After merge, update reference in mbed-os-example-ble sub-projects that use it.

### CODE REVIEW

File: BlueNrgHCIDriver.cpp [hunk 1]
```diff
@@ -555,7 +552,7 @@ class TransportDriver : public cordio::CordioHCITransportDriver {
     void spi_read_cb() {
         uint8_t data_buffer[256];
         while(true) {
-            _spi_read_sem.wait();
+            _spi_read_sem.try_acquire();
```
Comment (pan-) [inline BlueNrgHCIDriver.cpp]: Why not use `acquire` here ? We're not doing anything if it fails.

Comment (hugueskamba) [reply BlueNrgHCIDriver.cpp]: [Done](https://github.com/ARMmbed/cordio-ble-x-nucleo-idb0xa1/compare/feabf5cc58e1acb627307f9d2f412b802bdf12b2..d934c48e9bf65e4bf3d3af0f37f63818bd2f5823).

File: BlueNrgHCIDriver.cpp [hunk 2]
```diff
@@ -132,10 +132,7 @@ class HCIDriver : public cordio::CordioHCIDriver
 
             case ACI_READ_CONFIG_DATA_OPCODE:
                 // note: will send the HCI command to send the random address
-                cordio::BLE::deviceInstance().getGap().setAddress(
```
Comment (pan-) [inline BlueNrgHCIDriver.cpp]: We should keep this call, it is the only way - at the moment - to send a random static address. Enabling privacy would set non static random addresses.

Comment (hugueskamba) [reply BlueNrgHCIDriver.cpp]: [Done](https://github.com/ARMmbed/cordio-ble-x-nucleo-idb0xa1/compare/d934c48e9bf65e4bf3d3af0f37f63818bd2f5823..6c8e5b758d3d1db1f6f49c3bc67bfc8b32313b8a).
Thanks.

**Pull request merged**